### ISSUE TITLE
US data missing prior to 2020-05-24

### ISSUE THREAD

**Issue by jzohrab** [covidatlas/li#302]
Noted by @rtwfroody in https://github.com/covidatlas/li/issues/284, many US states are missing data in the generated reports.  e.g., from Li-reports-analysis (at https://docs.google.com/spreadsheets/d/1Xeq7nJdEbXtYpDR9LlXva2pZJfjNJtonDn7vSKuTFX0/edit#gid=1895621508):

![image](https://user-images.githubusercontent.com/1637133/87085055-38ad7580-c1fd-11ea-9096-763cd770ff85.png)

Need to check that cache is present for the given sources contributing to each location, and ensure that it's actually scraped and put into the database.

**Comment by jzohrab** [comment]
Some production timeseries sources have never been called.  Below is "invokes" table for prod:

![image](https://user-images.githubusercontent.com/1637133/87097712-c8f6b500-c213-11ea-8ab0-a3d187298ed0.png)

Adding some test cases and coverage for that.  I think the regeneration query is missing some cases.  In addition, we should perhaps regen some other sources as well, not just timeseries ... will consider.

**Comment by jzohrab** [comment]
The "regenerate" code isn't getting called correctly for timeseries sources.  Merged PR #306 to staging, will monitor.

Currently staging invokes table looks like this:

```
"type (S)","key (S)","lastInvoke (S)"
"regenerate","au-nsw","2020-07-10T08:41:07.898Z"
"regenerate","au-wa","2020-07-10T04:41:07.937Z"
"regenerate","be","2020-07-10T06:41:07.826Z"
"regenerate","ca","2020-07-10T10:41:07.829Z"
"regenerate","ca-ns","2020-07-10T12:41:07.924Z"
"regenerate","ch","2020-07-10T14:41:07.676Z"
"regenerate","cn","2020-07-10T02:41:07.576Z"
"regenerate","cy",
"regenerate","cz",
"regenerate","de",
"regenerate","ee",
"regenerate","es",
"regenerate","fr",
"regenerate","hk",
"regenerate","ie",
"regenerate","it",
"regenerate","jhu",
"regenerate","jhu-usa",
"regenerate","jp",
"regenerate","nl",
"regenerate","nyt",
"regenerate","sa",
"regenerate","se",
"regenerate","si",
"regenerate","ua",
"regenerate","us-az",
"regenerate","us-ca-mercury-news",
"regenerate","us-ca-sonoma-county",
"regenerate","us-nv-washoe-county",
"regenerate","us-ny",
"regenerate","us-va",
"regenerate","vn",
"regenerate","za",

```

I've bumped up the regen frequency, we should start to see some of the other countries getting filled in.  If it works, I'll tag and promote to production.

**Comment by jzohrab** [comment]
Nice, things starting to fill in on staging:

```
"type (S)","key (S)","lastInvoke (S)"
"regenerate","au-nsw","2020-07-10T08:41:07.898Z"
"regenerate","au-wa","2020-07-10T04:41:07.937Z"
"regenerate","be","2020-07-10T06:41:07.826Z"
"regenerate","ca","2020-07-10T10:41:07.829Z"
"regenerate","ca-ns","2020-07-10T12:41:07.924Z"
"regenerate","ch","2020-07-10T14:41:07.676Z"
"regenerate","cn","2020-07-10T02:41:07.576Z"

"regenerate","cy","2020-07-10T17:24:47.046Z"
"regenerate","cz","2020-07-10T17:39:46.984Z"
"regenerate","de","2020-07-10T17:54:47.057Z"
"regenerate","ee","2020-07-10T18:09:47.079Z"
"regenerate","es","2020-07-10T18:24:46.999Z"
"regenerate","fr","2020-07-10T18:39:46.988Z"
"regenerate","hk","2020-07-10T18:54:47.080Z"
"regenerate","ie","2020-07-10T19:09:48.086Z"
"regenerate","it","2020-07-10T19:24:47.047Z"
"regenerate","jhu","2020-07-10T19:39:47.096Z"
"regenerate","jhu-usa","2020-07-10T19:54:46.894Z"
"regenerate","jp","2020-07-10T20:09:47.056Z"
"regenerate","nl","2020-07-10T20:24:46.789Z"
"regenerate","nyt","2020-07-10T20:39:47.049Z"
"regenerate","sa","2020-07-10T20:54:47.200Z"
"regenerate","se","2020-07-10T21:09:46.908Z"
"regenerate","si","2020-07-10T21:24:46.811Z"
"regenerate","ua","2020-07-10T21:39:46.938Z"
"regenerate","us-az",
"regenerate","us-ca-mercury-news",
"regenerate","us-ca-sonoma-county",
"regenerate","us-nv-washoe-county",
"regenerate","us-ny",
"regenerate","us-va",
"regenerate","vn",
"regenerate","za",
```

**Comment by jzohrab** [comment]
Staging is now completely regenerated.  Production is in process (currently running jhu).  Will re-analyze reports when all timeseries are refreshed.

**Comment by jzohrab** [comment]
Prod invokes fully done, data filled in until mid-March which is great.

![image](https://user-images.githubusercontent.com/1637133/87225410-c652a780-c35a-11ea-8634-90f43b9e3f60.png)

The jhu-usa data goes all the way back to 2020-01-22, so I've moved that start date back to that date, and launched to prod.  That should fill in still more data, though it's of lower priority than other state sources if such exist.  Closing this.
